<?xml version="1.0"?>
<robot name="gopigo3">
  <ros2_control name="gpg_remote" type="system">
    <hardware>
      <plugin>gpg_remote/GPGRemoteHardware</plugin>
      <param name="host">192.168.0.201</param>
      <param name="port">8002</param>
    </hardware>
    
    <joint name="left_wheel">
      <command_interface name="velocity"/>
      <state_interface name="position"/>
      <state_interface name="velocity"/>  
    </joint>
    <joint name="right_wheel">
      <command_interface name="velocity"/>
      <state_interface name="position"/>
      <state_interface name="velocity"/>  
    </joint>
    <joint name="servo">
      <command_interface name="position"/>
      <state_interface name="position"/>
    </joint>
    <sensor name="line">
       <state_interface name="level0"/>
       <state_interface name="level1"/>
       <state_interface name="level2"/>
       <state_interface name="level3"/>
       <state_interface name="level4"/>
    </sensor>
    <sensor name="battery">
       <state_interface name="voltage"/>
    </sensor>
  </ros2_control>
  
  <link name="base_link">
  </link>
  
  <link name="right_wheel_link">
  </link>

  <joint name="right_wheel" type="continuous">
    <parent link="base_link"/>
    <child link="right_wheel_link"/>
  </joint>
  
  <link name="left_wheel_link">
  </link>

  <joint name="left_wheel" type="continuous">
    <parent link="base_link"/>
    <child link="left_wheel_link"/>
  </joint>
  
  <link name="servo_link">
  </link>
  
  <joint name="servo" type="revolute">
    <parent link="base_link"/>
    <child link="servo_link"/>
    <limit effort="1000.0" lower="-3.14" upper="3.14" velocity="1000"/>
  </joint>
  
</robot>
